tests/ci/cdk/pipeline/scripts/util.sh

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+set -ex
+
+if [[ -z "${PIPELINE_EXECUTION_ID:+x}" ]]; then
+  TRIGGER_TYPE="manual"
+else
+  TRIGGER_TYPE="pipeline"
+fi
+
+function assume_role() {
+  if [[ -z ${CROSS_ACCOUNT_BUILD_ROLE_ARN} ]]; then
+    echo "No role arn provided"
+    return 1
+  fi
+
+  local session_name=${CROSS_ACCOUNT_BUILD_SESSION:-"build-session"}
+  CREDENTIALS=$(aws sts assume-role --role-arn "${CROSS_ACCOUNT_BUILD_ROLE_ARN}" --role-session-name "${session_name}")
+  export AWS_ACCESS_KEY_ID=$(echo $CREDENTIALS | jq -r .Credentials.AccessKeyId)
+  export AWS_SECRET_ACCESS_KEY=$(echo $CREDENTIALS | jq -r .Credentials.SecretAccessKey)
+  export AWS_SESSION_TOKEN=$(echo $CREDENTIALS | jq -r .Credentials.SessionToken)
```

Comment:
I think this might cause the credentials to get logged? Should we call `set +x` when entering this function, and then re-enabling it on exit?